src/test/java/core/basesyntax/service/ReportWriterServiceTest.java

```diff
@@ -0,0 +1,79 @@
+package core.basesyntax.service;
+
+import core.basesyntax.service.impl.ReportWriterServiceImpl;
+import java.io.FileInputStream;
+import java.io.IOException;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class ReportWriterServiceTest {
+    private static final String READER_FAILURE_MESSAGE = "Cannot read from file {%s}!";
+    private static final String REPORT_FILE_NAME = "report.csv";
+    private static final String EMPTY_STRING = "";
+    private static final int READ_END_INDEX = -1;
+    private static ReportWriterService reportWriterService;
+
+    @BeforeAll
+    public static void setUp() {
+        reportWriterService = new ReportWriterServiceImpl();
+    }
+
+    @Test
+    public void write_correctInputData_Ok() {
+        StringBuilder reportStringBuilder = new StringBuilder("fruit,quantity");
+        reportStringBuilder.append(System.lineSeparator()).append("banana,152")
+                .append(System.lineSeparator()).append("apple,90")
+                .append(System.lineSeparator());
+        String report = reportStringBuilder.toString();
+
+        reportWriterService.writeReport(report, REPORT_FILE_NAME);
+
+        String actual = readReportFile();
+        Assertions.assertEquals(report, actual);
+    }
+
+    @Test
+    public void write_nullReportData_notOk() {
+        Assertions.assertThrows(RuntimeException.class,
+                () -> reportWriterService.writeReport(null, REPORT_FILE_NAME));
+    }
+
+    @Test
+    public void write_nullReportFileName_notOk() {
+        Assertions.assertThrows(RuntimeException.class,
+                () -> reportWriterService.writeReport(EMPTY_STRING, null));
+    }
+
+    @Test
+    public void write_nullInputFields_notOk() {
+        Assertions.assertThrows(RuntimeException.class,
+                () -> reportWriterService.writeReport(null, null));
+    }
+
+    @Test
+    public void write_emptyInputFields_notOk() {
+        Assertions.assertThrows(RuntimeException.class,
+                () -> reportWriterService.writeReport(EMPTY_STRING, EMPTY_STRING));
+    }
+

```

Comment:
use assertAll. It will look way better
![image](https://github.com/mate-academy/jv-fruit-shop-tests/assets/103361418/6546c42b-9203-42ff-af60-60a9f362686d)